sklearnex/linear_model/incremental_linear.py

```diff
@@ -0,0 +1,387 @@
+            "copy_X": ["boolean"],
+            "n_jobs": [Interval(numbers.Integral, -1, None, closed="left"), None],
+            "batch_size": [Interval(numbers.Integral, 1, None, closed="left"), None],
+        }
+
+    def __init__(self, *, fit_intercept=True, copy_X=True, n_jobs=None, batch_size=None):
+        self.fit_intercept = fit_intercept
+        self.copy_X = copy_X
+        self.n_jobs = n_jobs
+        self.batch_size = batch_size
+
+    def _onedal_supported(self, method_name, *data):
+        patching_status = PatchingConditionsChain(
+            f"sklearn.covariance.{self.__class__.__name__}.{method_name}"
+        )
+        return patching_status
+
+    _onedal_cpu_supported = _onedal_supported
+    _onedal_gpu_supported = _onedal_supported
+
+    def _onedal_predict(self, X, queue=None):
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        if sklearn_check_version("1.0"):
+            X = self._validate_data(
+                X,
+                dtype=[np.float64, np.float32],
+                copy=self.copy_X,
+            )
+        else:
+            X = check_array(
+                X,
+                dtype=[np.float64, np.float32],
+                copy=self.copy_X,
+            )
+
+        assert hasattr(self, "_onedal_estimator")
+        if self._need_to_finalize:
+            self._onedal_finalize_fit()
+        return self._onedal_estimator.predict(X, queue)
+
+    def _onedal_partial_fit(self, X, y, queue=None):
+        first_pass = not hasattr(self, "n_samples_seen_") or self.n_samples_seen_ == 0
+
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        if sklearn_check_version("1.0"):
+            X, y = self._validate_data(
+                X,
+                y,
+                dtype=[np.float64, np.float32],
+                reset=first_pass,
+                copy=self.copy_X,
+                multi_output=True,
+            )
+        else:
+            X = check_array(
+                X,
+                dtype=[np.float64, np.float32],
+                copy=self.copy_X,
+            )
+            y = check_array(
+                y,
+                dtype=[np.float64, np.float32],
+                copy=False,
+                ensure_2d=False,
+            )
+
+        if first_pass:
+            self.n_samples_seen_ = X.shape[0]
+            self.n_features_in_ = X.shape[1]
+        else:
+            self.n_samples_seen_ += X.shape[0]
+        onedal_params = {"fit_intercept": self.fit_intercept, "copy_X": self.copy_X}
+        if not hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator = self._onedal_incremental_linear(**onedal_params)
+        self._onedal_estimator.partial_fit(X, y, queue)
+        self._need_to_finalize = True
+
+    def _onedal_finalize_fit(self):
+        assert hasattr(self, "_onedal_estimator")
+        is_good_for_onedal = self.n_samples_seen_ >= self.n_features_in_ + int(
```

Comment:
```suggestion
        is_underdetermined = self.n_samples_seen_ < self.n_features_in_ + int(
```

Reply:
the naming was inherited from linear regression implementation which had been done before my PRs. but I can rename if you insist. then it is better to update it in LinearRegression as well. and I have to add that we have created task to align C++ side behavior with sklearn on undetermined cases. when it will be done this flag should be deleted at all.

Reply:
I would leave it as is for now to be consistent with LinearRegression and update both cases when status of mentioned task become clear.

Reply:
#1804 should be the changes needed for LinearRegression. Let me know what you think.